CONTRIBUTING.md

```diff
@@ -0,0 +1,63 @@
+Contributing
+============
+We gratefully accept bug reports and contributions from the community. There are some requirements we need to fulfill in order to be able to integrate contributions:
+
+ - As with any open source project, contributions will be reviewed by the project team and community and may need some modifications to be accepted.
+ - The contribution should not break API or ABI, unless there is a real justification for that. If there is an API change, the contribution, if accepted, will be merged only when there will be a major release.
+
+Contributor License Agreement (CLA)
+-----------------------------------
+- All contributions, whether large or small require a Contributor's License Agreement (CLA) to be accepted. This is because source code can possibly fall under copyright law and we need your consent to share in the ownership of the copyright.
+- To accept the Contributor’s License Agreement (CLA), individual contributors can do this by creating an mbed account and [accepting the online agreement here with a click through](https://developer.mbed.org/contributor_agreement/). Alternatively, for contributions from corporations, or those that do not wish to create an mbed account, a slightly different agreement can be found [here](https://www.mbed.com/en/about-mbed/contributor-license-agreements/). This agreement should be signed and returned to ARM as described in the instructions given.
+
+Coding Standards
+----------------
+- We would ask that contributions conform to [our coding standards](https://tls.mbed.org/kb/development/mbedtls-coding-standards), and that contributions are fully tested before submission.
```

Comment:
We should detail on what we mean by fully tested here, providing a list of test commands and scripts that should execute successfully, and/or reference the "Continuous Integration" section below (however, the githooks do for example not include `test-ref-configs.pl`).

Reply:
Perhaps we should modify the githooks to include `test-ref-configs.pl`?

Reply:
@RonEld `test-ref-configs.pl` runs for a few minutes, and I'd prefer to not have such larger scripts in the githooks. @andresag01 @Patater What do you think? 
I'd prefer to expand the `Continuous Integration` section below to give an explicit list of the things the user should check himself before pushing, like running the tests, adding the githooks, and potentially running other scripts like `test-ref-configs.pl`.

Reply:
@hanno-arm If it's a matter of few minutes, I agree with you and we should add it to the CI. Anyway, it's a different topic unrelated to this PR

Reply:
@RonEld I didn't mean to expand the CI, I meant to expand the CI section (or another one that might be more suitable) to include an explicit list of (the things the CI does too and) what _the user_ is supposed to do before opening the PR.

Reply:
@hanno-arm Actually, I think we should expand the CI
I expect the user to run the CI tests, but nonetheless, we should catch this if they don't run the tests